protostar/commands/test/test_scheduler.py

```diff
@@ -42,6 +50,11 @@ def run(
                 for test_suite in test_collector_result.test_suites
             ]
 
+            # A test case was broken
+            if exit_first and test_results_queue.failed():
+                self._live_logger.exit_before_log(test_collector_result)
+                return
```

Comment:
I'd move this check out of `TestLogger` to the `TestCommand` because at this point broken results come from `TestCollector`. You can skip instantiating  `TestScheduler`.

Reply:
It has to be here because we need the logger instanced to make the output look good - it returns instantly anyway.